Include/Library/OcCryptoLib.h

```diff
@@ -19,13 +19,83 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 #ifndef OC_CRYPTO_LIB_H
 #define OC_CRYPTO_LIB_H
 
+#include <Library/OcGuardLib.h>
+
+///
+/// A BIGNUM word. This is at best an integer of the platform's natural size
+/// to optimize memory accesses and arithmetic operation count.
+///
+typedef UINTN OC_BN_WORD;
```

Comment:
Hmm, is that ok to leave all these big num type definitions here? I thought that stuff like BigNum words should go in OcBigNumLib header, which should be included by OcCryptoLib header, and which should be the dependency of OcCryptoLib. It probably is ok, if we leave BigNumLib as an internal part of the OcCryptoLib, but does it really make sense?

Reply:
We do not have a dedicated BigNum lib, we have a BigNum family of functions inside OcCryptoLib. Some of the interfaces, as you mentioned below, are not intuitive, some restrictions (e.g. equal operand size) are seemingly arbitrary, and the interfaces do not exceed the exact amount and type of functions we need. The distinction between "public BN" and "private BN" is entirely artifical to make it easier to get an overview of the function's usage scope, "public BN" is still "private(!) OcCryptoLib".
However, for the actual point, I think the types should be private now, I had them public before because OcAppleKeysLib used to delcare its keys as BN, unless I realised it shares the same format with the Vault key and unified them again.

Reply:
Agreed.